### PR TITLE
Issue #739 XmlException when loading FederationMetadata.xml

### PR DESCRIPTION
**Pull request by lovemaths**

### CODE REVIEW

**Comment by brentschmaltz** [comment]
@lovemaths please add tests that read ADFS v2,v3,v4 similar to windows-ad-metadata.xml.

**Comment by lovemaths** [comment]
@brentschmaltz added the tests

File: test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/WsFederationConfigurationRetrieverTests.cs [hunk 1]
```diff
@@ -182,6 +192,42 @@ public static TheoryData<WsFederationMetadataTheoryData> ReadMetadataTheoryData
...
+                    new WsFederationMetadataTheoryData
+                    {
+                        Metadata = ReferenceMetadata.AdfsV3Metadata,
+                        SigingKey = ReferenceMetadata.AdfsV3MetadataSigningKey,
+                        Configuration = ReferenceMetadata.AdfsV3Endpoint
+                    },
+                    new WsFederationMetadataTheoryData
+                    {
+                        Metadata = ReferenceMetadata.AdfsV4Metadata,
+                        SigingKey = ReferenceMetadata.AdfsV4MetadataSigningKey,
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/WsFederationConfigurationRetrieverTests.cs]: We need a TestId

File: test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/adfs-v2-metadata.xml [hunk 2]
```diff
@@ -0,0 +1 @@
...as.microsoft.com/ws/2008/06/identity/claims/denyonlyprimarygroupsid" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri" FriendlyName="Deny only primary group SID" xmlns="urn:oasis:names:tc:SAML:2.0:assertion"/><Attribute Name="http://schemas.microsoft.com/ws/2008/06/identity/claims/groupsid" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri" FriendlyName="Group SID" xmlns="urn:oasis:names:tc:SAML:2.0:assertion"/><Attribute Name="http://schemas.microsoft.com/ws/2008/06/identity/claims/primarygroupsid" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri" FriendlyName="Primary group SID" xmlns="urn:oasis:names:tc:SAML:2.0:assertion"/><Attribute Name="http://schemas.microsoft.com/ws/2008/06/identity/claims/primarysid" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri" FriendlyName="Primary SID" xmlns="urn:oasis:names:tc:SAML:2.0:assertion"/><Attribute Name="http://schemas.microsoft.com/ws/2008/06/identity/claims/windowsaccountname" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri" FriendlyName="Windows account name" xmlns="urn:oasis:names:tc:SAML:2.0:assertion"/></IDPSSODescriptor><ContactPerson contactType="support"/></EntityDescriptor>
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/adfs-v2-metadata.xml]: Why have these in files and not strings?

Comment (lovemaths) [reply test/Microsoft.IdentityModel.Protocols.WsFederation.Tests/adfs-v2-metadata.xml]: We already have these files as strings in referenceMetadata. I thought you want tests for reading files too.

**Comment by brentschmaltz** [comment]
add the testid and then :shipit:

**Pull request merged**